include/zephyr/net/wifi_mgmt.h

```diff
@@ -962,6 +974,26 @@ struct wifi_dpp_params {
 	};
 };
 
+#define WIFI_WPS_PIN_MAX_LEN 8
+
+/** get/set operation for wps pin*/
+enum wifi_wps_pin_op {
+	/** Get operation */
+	WIFI_WPS_PIN_GET = 0,
+	/** Set operation */
+	WIFI_WPS_PIN_SET = 1,
+};
+
+/** Wi-Fi wps pin setup */
+struct wifi_wps_pin_params {
+	/** wps pin operation */
+	enum wifi_wps_pin_op oper;
+	/** get pin value*/
+	char get_pin[WIFI_WPS_PIN_MAX_LEN + 1];
+	/** set pin value*/
+	char set_pin[WIFI_WPS_PIN_MAX_LEN + 1];
```

Comment:
Can't we merge them as one `pin[WIFI_WPS_PIN_MAX_LEN + 1]`?  depending on the operation `pin` can either be input or output?

Reply:
Changed as comment